accessible/tree.service.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Service that handles all tree keyboard navigation.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.TreeService = ng.core
+  .Class({
+    constructor: function() {
+      console.log('making a new tree service');
+      this.activeDesc_ = {};
+      this.previousKey_;
+      this.trees = document.getElementsByClassName('blocklyTree');
+    },
+    createId: function(obj) {
+      if (obj && obj.id) {
+        return obj.id;
+      }
+      return 'blockly-' + Blockly.genUid();
+    },
+    setActiveAttribute: function(tree) {
```

Comment:
setAriaActiveDescendantAttribute

Still, same question as usual: can this be set directly in the template?